system/lib/pthread/library_pthread.c

```diff
@@ -883,7 +891,18 @@ void EMSCRIPTEN_KEEPALIVE emscripten_async_queue_on_thread_(
   // returning here, and it is the callee's responsibility to free up the memory after the call has
   // been performed.
   q->calleeDelete = 1;
-  emscripten_async_queue_call_on_thread(targetThread, q);
+  // The maybe_async() function will not be async on the same thread; force
```

Comment:
does `maybe_async` need to be updated here too?

Reply:
Thanks, yes, updating.